runtime/vm/JFRChunkWriter.cpp

```diff
@@ -815,4 +819,83 @@ VM_JFRChunkWriter::writeInitialSystemPropertyEvents(J9JavaVM *vm)
 	}
 }
 
+U_8 *
+VM_JFRChunkWriter::writeThreadDumpEvent()
+{
+	/* reserve size field */
+	U_8 *dataStart = _bufferWriter->getAndIncCursor(sizeof(U_32));
+
+	_bufferWriter->writeLEB128(ThreadDumpID);
+
+	/* write start time */
+	_bufferWriter->writeLEB128(j9time_current_time_millis());
+
+	char *result = (char *)j9mem_allocate_memory(sizeof(char) * THREAD_DUMP_EVENT_SIZE, OMRMEM_CATEGORY_VM);
+
+	if (NULL != result) {
```

Comment:
does JFR thread has vm access? I'm having the assertion failure inside `acquireExclusiveVMAccess`

Reply:
You wont know which thread is calling this code, it could be any java thread. You need to check if it has VM access first, if not acquire it. 

ou can check if it has vmacces with something like

```
if (vmThread->publicFlags & J9_PUBLIC_FLAGS_VM_ACCESS) {
```

You can only acquire exclusive access once you have vmaccess